src/test/clojure/xtdb/pgwire_test.clj

```diff
@@ -1982,8 +1985,36 @@ ORDER BY t.oid DESC LIMIT 1"
 
         (t/is (= [{"_id" 2, "f" 2.0, "i" 2} {"_id" 1, "f" 1.0, "i" 1}] (rs->maps rs)))))))
 
+(deftest test-primitive-array
+  (t/testing "primitive array"
+    (with-open [conn (jdbc-conn)]
+      (jdbc/execute! conn ["INSERT INTO docs(_id) VALUES (1)"])
+      (jdbc/execute! conn ["INSERT INTO docs(_id) VALUES (2)"])
+
+      (with-open [stmt (.prepareStatement conn "SELECT ARRAY(SELECT _id FROM docs ORDER BY _id) AS ids")]
+        (with-open [rs (.executeQuery stmt)]
+          (t/is (= [{"ids" "_int8"}]
+                   (result-metadata stmt)
+                   (result-metadata rs)))
+
+          (t/is (= [{"ids" [1 2]}] (rs->maps rs))))))))
+
+(deftest test-complex-array
```

Comment:
nit: I'd probably call this `test-elixir-client-complex-array` and lose the `t/testing`?